framework/src/play/plugins/PluginCollection.java

```diff
@@ -380,6 +411,35 @@ public void updatePlayPluginsList() {
     public List<PlayPlugin> getEnabledPlugins() {
         return enabledPlugins_readOnlyCopy;
     }
+
+    /**
+     * Returns new readonly list of all enabled plugins that define filters.
+     * @return List of plugins
+     */
+    public List<PlayPlugin> getEnabledPluginsWithFilters() {
+        return enabledPluginsWithFilters_readOnlyCopy;
+    }
+
+    @SuppressWarnings("unchecked")
+    public<T> F.Option<PlayPlugin.Filter<T>> composeFilters()
+    {
+      //Copy list of plugins here in case the list changes in the midst of doing composition...
+      //(Is it really necessary to do this?)
+      final List<PlayPlugin> pluginsWithFilters = new ArrayList<PlayPlugin>(this.getEnabledPluginsWithFilters());
+
+      if (pluginsWithFilters.isEmpty()) {
+        return F.Option.None();
+      } else {
+        final Iterator<PlayPlugin> itr = getEnabledPluginsWithFilters().iterator();
```

Comment:
Should this be using the cached list "pluginsWIthFilters" rather than calling "getEnabledPluginsWithFilters()" as per the comment above?